client/my-sites/current-site/notice.jsx

```diff
@@ -90,10 +93,64 @@ export class SiteNotice extends Component {
 		);
 	}
 
+	trialUpgradeNotice() {
+		const { translate, site, isTrialSite, currentPlan } = this.props;
+		const { expiry } = currentPlan || {};
+
+		if ( ! isTrialSite || ! expiry ) {
+			return null;
+		}
+
+		const eventProperties = { cta_name: 'migration-trial-upgrade-sidebar' };
+		const momentParams = { endsAt: currentPlan.expiry };
+		const daysRemaining = this.daysRemaining( momentParams );
+
+		const bannerText = this.promotionEndsToday( momentParams )
```

Comment:
For trials already ended, should we say something like "Your trial has already ended" instead of having minus day counts here?
![Screen Shot 2023-08-08 at 3 49 47 PM](https://github.com/Automattic/wp-calypso/assets/4074459/d6205f7c-41cd-4fec-82df-46b33c45c646)

Reply:
Of course. I have this on my list as a separate task, but being part of this PR makes sense.

Thanks for pointing this out.